test_hide_modules.py

```diff
@@ -62,6 +62,26 @@ def testDecoratorOnTest(self):
         self.assertRaises(ImportError, import_csv)
         self.assertRaises(ImportError, import_wsgiref_handlers)
 
+    def testRestoringHiddenNumpy(self):
+        """
+        Numpy has a complicated internal import scheme that requires special
+        handling with python 2.7.
```

Comment:
As I commented in the bug report, this will happen with any package that import a submodule.  This test is fine for now, but I may replace it with one that uses a stdlib package (encodings seems to work); this would remove numpy as a dependency for testing.

Reply:
yes agreed, i tried to find a standard lib that would work but didn't look too hard.